File: 201902646/10828.cpp
#include <iostream>
#include <string>

struct stack {
    int arr[10001];
    int cnt;
    void init() {
        cnt = 0;
    }

    void push(int X) {
        arr[cnt] = X;
        cnt++;
    }

    int pop() {
        if (cnt == 0) return -1;
        int temp = arr[cnt-1];
        cnt--;
        return temp;
    }

    int size() {
        return cnt;
    }

    bool empty() {
        if (cnt == 0) return 1;
        else return 0;
    }

    int top() {
        if (cnt == 0) return -1;
        return arr[cnt-1];
    }
};

int main() {
    int N;
    std::cin >> N;
    stack st;
    st.init();

    while(N) {
        std::string cmd;
        int num;
        std::cin >> cmd;
        if(cmd == "push") {
            std::cin >> num;
            st.push(num);
        }
        if(cmd == "pop") {
            std::cout << st.pop() << std::endl;
        }
        if(cmd == "size") {
            std::cout << st.size() << std::endl;
        }
        if(cmd == "empty") {
            std::cout << st.empty() << std::endl;
        }
        if(cmd == "top") {
            std::cout << st.top() << std::endl;
        }
        N--;
    }
}

File: 201902693/10828.py
# 입력이 최대 10000줄이 들어오기 때문에 
# sys.stdin.readline을 이용하지 않으면 시간초과
import sys
input = sys.stdin.readline
n = int(input())

st = []
for i in range(n) :
    cmd = list(input().split())
    if cmd[0] == 'push' :
        st.append(int(cmd[1]))
    if cmd[0] == 'top' :
        if len(st) == 0 :
            print(-1)
        else :    
            print(st[-1])
    if cmd[0] == 'size' :
        print(len(st))
    if cmd[0] == 'empty' :
        if len(st) == 0 :
            print(1)
        else :
            print(0)
    if cmd[0] == 'pop' :
        if len(st) == 0 :
            print(-1)
        else :
            print(st.pop())
            
